bundle/edu.gemini.programexport.servlet/src/main/scala/edu/gemini/programexport/servlet/ProgramExportServlet.scala

```diff
@@ -0,0 +1,478 @@
+    EncodeJson { case (c, a) =>
+      ("partnerTime" := TimeUnit.SECONDS.toMillis(a.getPartnerAward.getSeconds)) ->:
+        ("programTime" := TimeUnit.SECONDS.toMillis(a.getProgramAward.getSeconds)) ->:
+        ("category" := c.getDisplayName) ->:
+        jEmptyObject
+    }
+
+
+  implicit val timeAccountAllocationEncodeJson: EncodeJson[TimeAcctAllocation] =
+    EncodeJson { t =>
+      t.getCategories.asScala.toList.map(c => (c, t.getAward(c))).filter(_._2 != TimeAcctAward.ZERO).map {
+        case (c, a) => timeAccountAwardEncodeJson(c, a)
+      }.asJson
+    }
+
+  implicit val timingWindowEncodeJson: EncodeJson[TimingWindow] =
+    EncodeJson { tw =>
+      ("repeat" := tw.getRepeat) ->:
+        ("period" := tw.getPeriod) ->:
+        ("duration" := tw.getDuration) ->:
+        ("start" := tw.getStart) ->:
+        jEmptyObject
+    }
+
+  implicit val spatialProfileEncodeJson: EncodeJson[SpatialProfile] =
+    EncodeJson {
+      case PointSource =>
+        ("type" := "pointSource") ->:
+          jEmptyObject
+      case UniformSource =>
+        ("type" := "uniformSource") ->:
+          jEmptyObject
+      case GaussianSource(fwhm) =>
+        ("fwhm" := fwhm) ->:
+          ("type" := "gaussianSource") ->:
+          jEmptyObject
+    }
+
+  implicit val spectralDistributionEncodeJson: EncodeJson[SpectralDistribution] =
+    EncodeJson {
+      case BlackBody(t) =>
+        ("temperature" := t) ->:
+          ("type" := "blackBody") ->:
+          jEmptyObject
+      case PowerLaw(idx) =>
+        ("index" := idx) ->:
+          ("type" := "powerLaw") ->:
+          jEmptyObject
+      case EmissionLine(wl, w, f, c) =>
+        ("continuum" := c.value) ->:
+          ("flux" := f.value) ->:
+          ("width" := w.value) ->:
+          ("wavelength" := wl.length.toMicrons) ->:
+          ("type" := "emissionLine") ->:
+          jEmptyObject
+      case UserDefinedSpectrum(_, _) =>
+        ("type" := "userDefined") ->:
+          jEmptyObject
+      case s: LibraryStar =>
+        ("sedSpectrum" := s.sedSpectrum) ->:
+          ("type" := "libraryStar") ->:
+          jEmptyObject
+      case n: LibraryNonStar =>
+        ("sedSpectrum" := n.sedSpectrum) ->:
+          ("label" := n.label) ->:
+          ("type" := "libraryNonStar") ->:
+          jEmptyObject
+      case _: AuxFileSpectrum =>
+        jEmptyObject
+    }
+
+  // Determine the target type.
+  def targetType(te: TargetEnvironment, spTarget: SPTarget): String =
+    if (te.getUserTargets.asScala.map(_.target).contains(spTarget))
+      te.getUserTargets.asScala.find(_.target == spTarget).map(_.`type`.displayName).get
+    else if (te.getGuideEnvironment.containsTarget(spTarget))
+      "GuideStar"
+    else
+      "Base"
+
+  // Unfortunately, we have to pass a lot of crap to this to get everything we need.
+  val siderealTargetEncodeJson: EncodeJson[(TargetEnvironment, SPTarget, SiderealTarget, Option[Int])] =
+    EncodeJson { case (te, sp, t, idx) =>
```

Comment:
Not saying it's *wrong* (I don't know) but for me at least, it's a bit unexpected to see the `EncodeJson` typeclass involved in this instead of a function:

`def encodeSiderealTarget(te: TargetEnvironment, sp: SPTarget, t: SiderealTarget, idx: Option[Int]): Json = ???`

The typeclass would be needed if it were going to be providing the definition of an encoder for some type implicitly?